src/app/icd/client/ICDClientStorage.h

```diff
@@ -61,16 +61,31 @@ class ICDClientStorage
      * Store updated ICD ClientInfo to storage when ICD registration completes or check-in message
      * comes.
      *
-     * @param[in] aICDClientInfo the updated ICD Client Info.
+     * @param[in] clientInfo the updated ICD Client Info.
      */
     virtual CHIP_ERROR StoreEntry(const ICDClientInfo & clientInfo) = 0;
 
+    /**
+     * Remove ICD key from clientInfo when ICD registration fails
+     *
+     * @param[inout] clientInfo the updated ICD Client Info.
```

Comment:
Updated in what way?  The one implementation here is not updating anything about the clientinfo, as far as I can tell.